### PR TITLE
vc: Clean up directories in case MkdirAll fails

### LINKED ISSUE #2230: Clean up directories in case MkdirAll fails
# Description of problem
```
	for cid, cstate := range fs.containerState {
		cdir := filepath.Join(sandboxDir, cid)
		if err := os.MkdirAll(cdir, dirMode); err != nil {
			return err
```
In case subsequent MkdirAll call fails, the created directories would be left behind.

# Expected result

Previously created directories should be cleaned up.

# Actual result

### PR DESCRIPTION
**Pull request by tedyu**
Fixes #2230

Signed-off-by: Ted Yu <yuzhihong@gmail.com>

### CODE REVIEW

**Comment by tedyu** [comment]
@lifupan @grahamwhaley 
Please review.

File: virtcontainers/persist/fs/fs.go [hunk 1]
```diff
@@ -126,12 +126,25 @@ func (fs *FS) ToDisk(ss persistapi.SandboxState, cs map[string]persistapi.Contai
...
 	}
 
+	var dirCreationErr error
+	var createdDirs []string
+	defer func() {
+		if dirCreationErr != nil && len(createdDirs) > 0 {
+			for _, dir := range createdDirs {
+				// remove the persistFile first
+				cfile := filepath.Join(dir, persistFile)
+				os.Remove(cfile)
```
Comment (lifupan) [inline virtcontainers/persist/fs/fs.go]: Is it better to merge those two removes into one step of os.RemoveAll ?

Comment (tedyu) [reply virtcontainers/persist/fs/fs.go]: @lifupan 
Your comment has been addressed.

**Comment by tedyu** [comment]
@lifupan @grahamwhaley
Can you trigger test run for ubuntu ?

**Comment by tedyu** [comment]
@grahamwhaley 
Can you approve this ?

thanks

**Comment by tedyu** [comment]
@jodh-intel 
Can you approve this ?

thanks

**Comment by tedyu** [comment]
@amshinde 
Can you approve this ?

thanks

**Pull request merged**